src/core/server/saved_objects/migrations/core/migration_context.ts

```diff
@@ -199,6 +208,37 @@ export function disableUnknownTypeMappingFields(
   };
 }
 
+/**
+ * This function is used to modify the target mappings object by deleting specified type mappings fields.
+ *
+ * The function operates under the following conditions:
+ * - It checks if the 'migrations.delete.enabled' configuration is set to true.
+ * - If true, it retrieves the 'migrations.delete.types' configuration
+ * - For each type, it deletes the corresponding property from the target mappings object.
+ *
+ * The purpose of this function is to allow for dynamic modification of the target mappings object
+ * based on the application's configuration. This can be useful in scenarios where certain type
+ * mappings are no longer needed and should be removed from the target mappings.
+ *
+ * Note: The function does not return a value. It directly modifies the targetMappings object passed to it.
+ *
+ * @param {Object} targetMappings - The target mappings object to be modified.
+ * @param {Object} opensearchDashboardsRawConfig - The application's configuration object.
```

Comment:
this is object parsed from yaml config file?

Reply:
yes. i would need dig deeper into what they opted to use it this way as they have the config but mimicked what they did before.